model/fn-execution/src/main/proto/org/apache/beam/model/fn_execution/v1/beam_fn_api.proto

```diff
@@ -1063,6 +1087,10 @@ message StateAppendRequest {
   // Represents a part of a logical byte stream. Elements within
   // the logical byte stream are encoded in the nested context and
   // multiple append requests are concatenated together.
+  // For OrderedListState, elements of TimeStampedValue<T> should be encoded
+  // with TimestampedValueCoder.of(LengthPrefixCoder.of(Coder<T>)), so that
+  // the request handler knows how to decode timestamps from the data without
+  // decoding the value bytes.
```

Comment:
Perhaps note that data will be returned in the same format?

Reply:
This comment has been updated in the proto change only PR (#31092)